src/main/java/com/doctusoft/math/ExponentialDelays.java

```diff
@@ -0,0 +1,438 @@
+package com.doctusoft.math;
+
+import com.doctusoft.annotation.Beta;
+import com.doctusoft.java.Failsafe;
+
+import java.util.*;
+import java.util.function.*;
+import java.util.stream.*;
+
+import static java.util.Objects.*;
+
+/**
+ * This utility helps create streams of exponentially increasing int or long values complying with various criteria.
+ * Assembling all those criteria is done in a builder style defined by {@link OfInt} and {@link OfLong} internal
+ * interfaces. Invoking their {@code build()} method returns a new {@link IntStream} or {@link LongStream} instance
+ * with the desired attributes.
+ */
+@Beta
+public interface ExponentialDelays {
```

Comment:
Most of the time I want to maximize the amount of time spent on executing a callable.

For example: 
* I want a request to take at most 10 seconds. 
* I need to call two external systems. 
* The second call depends on the first so I can't do them concurrently. 

In this case I want to say something like: the first request should take at most 9 seconds.
The second should take at most the remaining time of the 10 seconds.

I can do something like this if I configure the ExponentialDelays properly, but it is not really convenient.
A limitOverallDelays method would make it easy.

A limitOverallExecution would be even better. For this you should generate the delay stream dynamically so you can consider the execution time of the task too and not just the delays.
Basically record the time time on build and stop the iteration if the next value would bring the overall time above the limit.